File: FindDisappearedNumber.swift
// Time Complexity : (O)n
// Space Complexity : (O)1
// Did this code successfully run on Leetcode : yes
//get abs index value of the array and multiply by -1.
//second iteration find the positive value index to get the missing number



class Solution {
    func findDisappearedNumbers(_ nums: [Int]) -> [Int] {
        //edge
        if nums.isEmpty {
            return nums
        }
        //logic
        let length = nums.count
        var numbers = nums
        //go to index of the array and get abs value and multiply by -1 for the value
        for i in 0..<length {
            let valueIndex = (abs(numbers[i])-1)
            numbers[valueIndex] = abs(numbers[valueIndex]) * -1
        }
        //result
        var result = [Int]()
        for i in 0..<length {
            if numbers[i] > 0 {
              result.append(i+1)
            }
        }
        return result
    }
}

File: findDisappearednumber.java
// Time Complexity : o(n)
// Space Complexity : o(n)
// Did this code successfully run on Leetcode :
// Any problem you faced while coding this :


// Your code here along with comments explaining your approach


class Solution {
    public List<Integer> findDisappearedNumbers(int[] nums) {
        
        
        HashMap<Integer,Boolean> map = new HashMap<Integer,Boolean>();
        
        for(int i=0; i<nums.length; i++){
            map.put(nums[i],true);
        }
        
        List<Integer> result = new ArrayList<Integer>();
        
        for(int i=1; i<=nums.length; i++){
           
            if(!map.containsKey(i)){
                result.add(i);
            }
        }
        
        return result;
        
    }
}